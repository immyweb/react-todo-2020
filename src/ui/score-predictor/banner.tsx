import * as React from 'react';
import glamorous from 'glamorous';

const Banner = () => {
  return (
    <div>
      <Link href="https://www.sunbets.co.uk/sunsix?n=101&b=19079960&c=3&utm_source=22&utm_medium=3&utm_content=%3Fpayload%3Dfootball_predictor&utm_campaign=101">
        <Image
          src="https://www.thesun.co.uk/wp-content/uploads/2018/03/pod-fixtures-fullbanner1.png"
          alt="Get six scores right &amp; WIN Â£1 MILLION"
        />
      </Link>
    </div>
  );
};

const Link = glamorous.a({
  label: 'banner-link',
  display: 'block',
});

const Image = glamorous.img({
  label: 'banner-image',
  width: '100%',
  marginBottom: 25,
});

export default Banner;
