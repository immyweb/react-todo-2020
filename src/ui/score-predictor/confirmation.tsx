import * as React from 'react';

import { colours } from '../../styles/colours';

import Button from '../buttons/buttons';
import { MessageContainer, Heading, Graphic, Copy } from './styles';

interface IProps {
  slideWidth: number;
  onConfirm(): void;
}

export const Confirmation: React.SFC<IProps> = ({ slideWidth, onConfirm }) => {
  return (
    <MessageContainer width={slideWidth}>
      <div>
        <Graphic src="https://www.thesun.co.uk/wp-content/uploads/2018/03/celebrate.png" />
        <Heading fontColour={colours.$football}>NICE ONE!</Heading>
        <Copy>You're ready to go!</Copy>
        <Button background={colours.$football} onClick={onConfirm}>
          See final table
        </Button>
      </div>
    </MessageContainer>
  );
};
