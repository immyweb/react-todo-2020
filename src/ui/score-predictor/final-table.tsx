import * as React from 'react';

import { EndScreen } from './end-screen';

import { ITeam, TableType } from './types';
import { FixtureTable, THead, THeading, THBorder, TrPrem, TrChamp, TdTeam, Badge, Headline } from './styles';
import { fonts } from '../../styles/fonts';

interface IProps {
  table: ITeam[];
  timer: number;
  exitUrl: string;
  tableType: string;
}

interface IState {
  showEndScreen: boolean;
}

export class FinalTable extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      showEndScreen: false,
    };
  }

  componentDidMount() {
    // Display End Screen after 10 seconds
    setTimeout(() => {
      this.setState({ showEndScreen: true });
    }, this.props.timer);
  }

  onGoToUrl = () => {
    window.location.assign(this.props.exitUrl);
  };

  onCloseScreen = () => {
    this.setState({ showEndScreen: false });
  };

  // Preact currently does not support fragments.
  // See https://github.com/developit/preact/issues/946
  // Please refactor this, when support for this feature is enabled.

  // renderContent(result: ITeam, i: number) {
  // 	return (
  // 		<>
  // 			<TdTeam definedWidth={20} fontName={fonts.$Roboto}>
  // 				{i + 1}
  // 			</TdTeam>
  // 			<TdTeam definedWidth={45}>
  // 				<Badge alt={result.team} src={result.badge} />
  // 			</TdTeam>
  // 			<TdTeam fontAlignment="left">{result.team}</TdTeam>
  // 			<TdTeam fontName={fonts.$Roboto}>{result.played}</TdTeam>
  // 			<TdTeam fontName={fonts.$Roboto}>{result.win}</TdTeam>
  // 			<TdTeam fontName={fonts.$Roboto}>{result.draw}</TdTeam>
  // 			<TdTeam fontName={fonts.$Roboto}>{result.lose}</TdTeam>
  // 			<TdTeam>{result.points}</TdTeam>
  // 		</>
  // 	);
  // }

  renderRows(table: ITeam[], tableType: string) {
    return table.map((result, i) => {
      if (tableType === TableType.Prem) {
        return (
          <TrPrem key={result.team} definedHeight={49}>
            <TdTeam definedWidth={20} fontName={fonts.$Roboto}>
              {i + 1}
            </TdTeam>
            <TdTeam definedWidth={45}>
              <Badge alt={result.team} src={result.badge} />
            </TdTeam>
            <TdTeam fontAlignment="left">{result.team}</TdTeam>
            <TdTeam fontName={fonts.$Roboto}>{result.played}</TdTeam>
            <TdTeam fontName={fonts.$Roboto}>{result.win}</TdTeam>
            <TdTeam fontName={fonts.$Roboto}>{result.draw}</TdTeam>
            <TdTeam fontName={fonts.$Roboto}>{result.lose}</TdTeam>
            <TdTeam>{result.points}</TdTeam>
          </TrPrem>
        );
      }
      if (tableType === TableType.Champ) {
        return (
          <TrChamp key={result.team} definedHeight={49}>
            <TdTeam definedWidth={20} fontName={fonts.$Roboto}>
              {i + 1}
            </TdTeam>
            <TdTeam definedWidth={45}>
              <Badge alt={result.team} src={result.badge} />
            </TdTeam>
            <TdTeam fontAlignment="left">{result.team}</TdTeam>
            <TdTeam fontName={fonts.$Roboto}>{result.played}</TdTeam>
            <TdTeam fontName={fonts.$Roboto}>{result.win}</TdTeam>
            <TdTeam fontName={fonts.$Roboto}>{result.draw}</TdTeam>
            <TdTeam fontName={fonts.$Roboto}>{result.lose}</TdTeam>
            <TdTeam>{result.points}</TdTeam>
          </TrChamp>
        );
      }
      return (
        <TrPrem key={result.team} definedHeight={49}>
          <TdTeam definedWidth={20} fontName={fonts.$Roboto}>
            {i + 1}
          </TdTeam>
          <TdTeam definedWidth={45}>
            <Badge alt={result.team} src={result.badge} />
          </TdTeam>
          <TdTeam fontAlignment="left">{result.team}</TdTeam>
          <TdTeam fontName={fonts.$Roboto}>{result.played}</TdTeam>
          <TdTeam fontName={fonts.$Roboto}>{result.win}</TdTeam>
          <TdTeam fontName={fonts.$Roboto}>{result.draw}</TdTeam>
          <TdTeam fontName={fonts.$Roboto}>{result.lose}</TdTeam>
          <TdTeam>{result.points}</TdTeam>
        </TrPrem>
      );
    });
  }

  render() {
    return (
      <div>
        <Headline fontName={fonts.$theSunHeavyCondensed}>Here's how YOU think the table will finish</Headline>
        <FixtureTable>
          <THead>
            <tr>
              <THeading />
              <THeading />
              <THeading borderRight={true} fontAlignment="left">
                Team
              </THeading>
              <THeading definedWidth={35}>
                Pl <THBorder>|</THBorder>
              </THeading>
              <THeading definedWidth={35}>
                W <THBorder>|</THBorder>
              </THeading>
              <THeading definedWidth={35}>
                D <THBorder>|</THBorder>
              </THeading>
              <THeading definedWidth={35}>
                L<THBorder>|</THBorder>
              </THeading>
              <THeading definedWidth={35}>Pts</THeading>
            </tr>
          </THead>
          <tbody>{this.renderRows(this.props.table, this.props.tableType)}</tbody>
        </FixtureTable>
        {this.state.showEndScreen && <EndScreen onCloseScreen={this.onCloseScreen} onGoToUrl={this.onGoToUrl} />}
      </div>
    );
  }
}
