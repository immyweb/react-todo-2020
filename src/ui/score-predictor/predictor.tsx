import * as React from 'react';
import Measure from 'react-measure';

import IntroScreen from './intro-screen';
import Banner from './banner';
import Controls from './controls';
import { Slides } from './slides';

import { IFixture, ITeam, ITally } from './types';

const playedMatches: any = [];

export const groupFixtures = (fixtures: IFixture[], groupNumber: number) => {
  const grouped = [];
  const fixturesCopy = fixtures.map(a => ({ ...a }));
  while (fixturesCopy.length > 0) {
    grouped.push(fixturesCopy.splice(0, groupNumber));
  }

  return grouped;
};

interface IProps {
  fixtures: IFixture[];
  teams: ITeam[];
  groupNumber: number;
  introHeadline: string;
  introTxt: string;
  onPredict(tally: ITally[]): void;
  onConfirm(): void;
  onLoaded(): void;
}

interface IState {
  intro: boolean;
  currentSlide: number;
  currentPos: number;
  slideWidth: number;
  totalWidth: number;
  slideCount: number;
  totalFixtures: number;
  fixturesPlayed: number;
  allFixturesCompleted: boolean;
  tables: IFixture[][];
}

export class Predictor extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      intro: true,
      currentSlide: 0,
      currentPos: 0,
      slideWidth: 0,
      totalWidth: 0,
      slideCount: 0,
      totalFixtures: 0,
      fixturesPlayed: 0,
      allFixturesCompleted: false,
      tables: [],
    };
  }

  componentDidMount() {
    const totalFixtures = this.props.fixtures.length;
    const fixturesGrouped = groupFixtures(this.props.fixtures, this.props.groupNumber);
    const tableCount = fixturesGrouped.length;

    this.setState({
      totalFixtures,
      slideCount: tableCount,
      tables: fixturesGrouped,
    });
  }

  onStartGame = () => {
    this.setState({
      intro: false,
    });
  };

  onTogglePrev = () => {
    const current = this.state.currentSlide;
    let prev = current - 1;

    if (prev < 0) {
      prev = 0;
    }

    const newPos = this.state.slideWidth * prev;

    this.setState({
      currentSlide: prev,
      currentPos: newPos,
    });
  };

  onToggleNext = () => {
    const current = this.state.currentSlide;
    let next = current + 1;

    if (next > this.state.slideCount - 1) {
      next = current;
    }

    const newPos = this.state.slideWidth * next;

    this.setState({
      currentSlide: next,
      currentPos: newPos,
    });
  };

  onPlayed = (fixtureId: number) => {
    if (!playedMatches.includes(fixtureId)) {
      playedMatches.push(fixtureId);
      this.setState({
        fixturesPlayed: playedMatches.length,
      });
    }
    if (playedMatches.length === this.state.totalFixtures) {
      this.setState(
        {
          allFixturesCompleted: true,
        },
        () => {
          setTimeout(() => {
            this.props.onConfirm();
          }, 500);
        },
      );
    }
  };

  render() {
    const { teams, onPredict, onLoaded, introHeadline, introTxt } = this.props;
    const {
      tables,
      currentSlide,
      slideCount,
      currentPos,
      totalWidth,
      slideWidth,
      intro,
    } = this.state;

    return (
      <Measure
        bounds
        onResize={contentRect => {
          const sWidth = contentRect.bounds!.width;
          const tWidth = sWidth * this.state.slideCount;
          this.setState({
            slideWidth: sWidth,
            totalWidth: tWidth,
          });
          setTimeout(() => {
            onLoaded();
          }, 0);
        }}
      >
        {({ measureRef }) => (
          <section ref={measureRef}>
            {intro && (
              <IntroScreen
                slideWidth={slideWidth}
                introHeadline={introHeadline}
                introTxt={introTxt}
                onStartGame={this.onStartGame}
              />
            )}

            <Banner />

            <Slides
              tables={tables}
              teams={teams}
              onPredict={onPredict}
              currentSlide={currentSlide}
              currentPos={currentPos}
              totalWidth={totalWidth}
              slideWidth={slideWidth}
              onPlayed={this.onPlayed}
              onToggleNext={this.onToggleNext}
            />

            <Controls
              onTogglePrev={this.onTogglePrev}
              onToggleNext={this.onToggleNext}
              currentSlide={currentSlide}
              slideCount={slideCount}
            />
          </section>
        )}
      </Measure>
    );
  }
}
