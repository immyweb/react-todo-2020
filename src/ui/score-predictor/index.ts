import { ScorePredictor } from '../../ui/score-predictor/score-predictor';

export default ScorePredictor;
