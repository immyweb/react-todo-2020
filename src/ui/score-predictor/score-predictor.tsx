import * as React from 'react';
import glamorous from 'glamorous';

import { FinalTable } from './final-table';
import { Predictor } from './predictor';

import { IFixture, ITeam, ITally } from './types';
import tableGenerator from './table-generator';

export interface IProps {
  teams: ITeam[];
  fixtures: IFixture[];
  introHeadline: string;
  introTxt: string;
  tableType: string;
}

interface IState {
  completed: boolean;
  loaded: boolean;
  scoreTally: ITally[];
  finalStanding: ITeam[];
  _fixtures: IFixture[];
}

interface IGProps {
  loaded?: boolean;
}

export class ScorePredictor extends React.Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      completed: false,
      loaded: false,
      scoreTally: [],
      finalStanding: [],
      _fixtures: [],
    };
  }

  UNSAFE_componentWillMount() {
    const tempFixtures = this.props.fixtures.map((data, i) => ({
      ...data,
      id: i,
    }));

    this.setState({
      _fixtures: tempFixtures,
    });
  }

  updateTallyState(tally: ITally[]) {
    this.setState({
      scoreTally: tableGenerator.addToTally(tally, this.state.scoreTally),
    });
  }

  finalise() {
    const finalStanding = tableGenerator.finalise(this.props.teams, this.state.scoreTally);
    this.setState({
      finalStanding,
      completed: true,
    });
  }

  onConfirm = () => {
    this.finalise();
  };

  onLoaded = () => {
    this.setState({
      loaded: true,
    });
  };

  render() {
    const { teams, introHeadline, introTxt, tableType } = this.props;
    const { _fixtures, finalStanding, loaded } = this.state;

    return (
      <Container loaded={loaded}>
        {!this.state.completed ? (
          <Predictor
            fixtures={_fixtures}
            teams={teams}
            groupNumber={8}
            introHeadline={introHeadline}
            introTxt={introTxt}
            onPredict={(tally) => this.updateTallyState(tally)}
            onConfirm={this.onConfirm}
            onLoaded={this.onLoaded}
          />
        ) : (
          <FinalTable
            table={finalStanding}
            timer={10000}
            tableType={tableType}
            exitUrl={
              'https://www.sunbets.co.uk/sunsix?n=101&b=19079960&c=3&utm_source=22&utm_medium=3&utm_content=%3Fpayload%3Dfootball_predictor&utm_campaign=101'
            }
          />
        )}
      </Container>
    );
  }
}

const Container = glamorous.div<IGProps>(
  {
    label: 'score-predictor-game',
    position: 'relative',
  },
  ({ loaded }) => ({
    height: loaded ? 'auto' : 660,
    overflow: loaded ? 'auto' : 'hidden',
  }),
);
