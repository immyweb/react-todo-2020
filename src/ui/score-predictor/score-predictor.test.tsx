import * as React from 'react';
import { shallow } from 'enzyme';
import { ScorePredictor } from './score-predictor';

import { Predictor } from './predictor';
import { FinalTable } from './final-table';

const teams = [
  {
    team: 'Liverpool',
    badge: 'https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67',
    played: 31,
    win: 18,
    draw: 9,
    lose: 4,
    points: 63,
  },
  {
    team: 'C Palace',
    badge:
      'https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72',
    played: 31,
    win: 7,
    draw: 9,
    lose: 15,
    points: 30,
  },
  {
    team: 'Brighton',
    badge:
      'https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96',
    played: 30,
    win: 8,
    draw: 10,
    lose: 12,
    points: 34,
  },
  {
    team: 'Leicester',
    badge:
      'https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90',
    played: 30,
    win: 10,
    draw: 10,
    lose: 10,
    points: 40,
  },
];

const fixtures = [
  { team1: 'C Palace', team2: 'Liverpool' },
  { team1: 'Brighton', team2: 'Leicester' },
  { team1: 'C Palace', team2: 'Liverpool' },
  { team1: 'Brighton', team2: 'Leicester' },
  { team1: 'C Palace', team2: 'Liverpool' },
  { team1: 'Brighton', team2: 'Leicester' },
  { team1: 'C Palace', team2: 'Liverpool' },
  { team1: 'Brighton', team2: 'Leicester' },
];

function shallowRender(props?: any) {
  return shallow(
    <ScorePredictor
      teams={teams}
      fixtures={fixtures}
      introHeadline={'EFL CHAMPIONSHIP PREDICTOR'}
      introTxt={
        'Predict Home Win, Draw or Away Win to see who will finish in those crucial Promotion and Relegation places.'
      }
      {...props}
    />,
  );
}

describe('<ScorePredictor />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('adds IDs to fixtures when componentWillMount', () => {
    const wrapper = shallowRender();
    expect(wrapper.state('_fixtures')[0]).toEqual({
      id: 0,
      team1: 'C Palace',
      team2: 'Liverpool',
    });
    expect(wrapper.state('_fixtures')[1]).toEqual({
      id: 1,
      team1: 'Brighton',
      team2: 'Leicester',
    });
  });

  it('renders <Predictor /> on launch', () => {
    const wrapper = shallowRender();
    const predicator = wrapper.find(Predictor);
    expect(predicator.length).toBe(1);
    const finalTable = wrapper.find(FinalTable);
    expect(finalTable.length).toBe(0);
  });

  it('renders <FinalTable /> when completed state is true', () => {
    const wrapper = shallowRender();
    wrapper.setState({ completed: true });
    const predicator = wrapper.find(Predictor);
    expect(predicator.length).toBe(0);
    const finalTable = wrapper.find(FinalTable);
    expect(finalTable.length).toBe(1);
  });

  it('addToTally updates scoreTally with correct value when existing prediction changes', () => {
    const wrapper = shallowRender();
    expect(wrapper.state('scoreTally')).toEqual([]);
    const instance = wrapper.instance() as any;
    const tally = [
      { id: 0, points: 3, resultType: 'WIN', team: 'Liverpool' },
      { id: 0, points: 0, resultType: 'LOSE', team: 'C Palace' },
    ];
    instance.updateTallyState(tally);
    expect(wrapper.state('scoreTally')).toEqual(tally);
    instance.updateTallyState(tally);
    expect(wrapper.state('scoreTally')).toEqual(tally);
    const tally2 = [
      { id: 0, points: 1, resultType: 'DRAW', team: 'Liverpool' },
      { id: 0, points: 1, resultType: 'DRAW', team: 'C Palace' },
    ];
    instance.updateTallyState(tally2);
    expect(wrapper.state('scoreTally')).toEqual(tally2);
    const tally3 = [
      { id: 0, points: 0, resultType: 'LOSE', team: 'Liverpool' },
      { id: 0, points: 3, resultType: 'WIN', team: 'C Palace' },
    ];
    instance.updateTallyState(tally3);
    expect(wrapper.state('scoreTally')).toEqual(tally3);
  });

  it('addToTally updates scoreTally with correct value when new prediction added', () => {
    const wrapper = shallowRender();
    expect(wrapper.state('scoreTally')).toEqual([]);
    const instance = wrapper.instance() as any;
    const tally = [
      { id: 0, points: 3, resultType: 'WIN', team: 'Liverpool' },
      { id: 0, points: 0, resultType: 'LOSE', team: 'C Palace' },
    ];
    instance.updateTallyState(tally);
    expect(wrapper.state('scoreTally')).toEqual(tally);
    const tally2 = [
      { id: 1, points: 1, resultType: 'DRAW', team: 'Brighton' },
      { id: 1, points: 1, resultType: 'DRAW', team: 'Leicester' },
    ];
    instance.updateTallyState(tally2);
    expect(wrapper.state('scoreTally')).toEqual([...tally, ...tally2]);
  });

  it('calling finalise will show final table', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as any;
    instance.finalise();
    wrapper.update();
    expect(wrapper.state('completed')).toBe(true);
    const finalTable = wrapper.find(FinalTable);
    expect(finalTable.length).toBe(1);
  });

  it('calling onConfirm will show final table', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as any;
    instance.onConfirm();
    wrapper.update();
    expect(wrapper.state('completed')).toBe(true);
    const finalTable = wrapper.find(FinalTable);
    expect(finalTable.length).toBe(1);
  });

  it('calling onLoaded will set loaded to true', () => {
    const wrapper = shallowRender();
    const instance = wrapper.instance() as any;
    instance.onLoaded();
    wrapper.update();
    expect(wrapper.state('loaded')).toBe(true);
  });
});

// win / lose / draw design
// final table design
