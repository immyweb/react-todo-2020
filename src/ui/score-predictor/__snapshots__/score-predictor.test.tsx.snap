// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScorePredictor /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScorePredictor
    fixtures={
      Array [
        Object {
          "team1": "C Palace",
          "team2": "Liverpool",
        },
        Object {
          "team1": "Brighton",
          "team2": "Leicester",
        },
        Object {
          "team1": "C Palace",
          "team2": "Liverpool",
        },
        Object {
          "team1": "Brighton",
          "team2": "Leicester",
        },
        Object {
          "team1": "C Palace",
          "team2": "Liverpool",
        },
        Object {
          "team1": "Brighton",
          "team2": "Leicester",
        },
        Object {
          "team1": "C Palace",
          "team2": "Liverpool",
        },
        Object {
          "team1": "Brighton",
          "team2": "Leicester",
        },
      ]
    }
    introHeadline="EFL CHAMPIONSHIP PREDICTOR"
    introTxt="Predict Home Win, Draw or Away Win to see who will finish in those crucial Promotion and Relegation places."
    teams={
      Array [
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
          "draw": 9,
          "lose": 4,
          "played": 31,
          "points": 63,
          "team": "Liverpool",
          "win": 18,
        },
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
          "draw": 9,
          "lose": 15,
          "played": 31,
          "points": 30,
          "team": "C Palace",
          "win": 7,
        },
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
          "draw": 10,
          "lose": 12,
          "played": 30,
          "points": 34,
          "team": "Brighton",
          "win": 8,
        },
        Object {
          "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
          "draw": 10,
          "lose": 10,
          "played": 30,
          "points": 40,
          "team": "Leicester",
          "win": 10,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Predictor
        fixtures={
          Array [
            Object {
              "id": 0,
              "team1": "C Palace",
              "team2": "Liverpool",
            },
            Object {
              "id": 1,
              "team1": "Brighton",
              "team2": "Leicester",
            },
            Object {
              "id": 2,
              "team1": "C Palace",
              "team2": "Liverpool",
            },
            Object {
              "id": 3,
              "team1": "Brighton",
              "team2": "Leicester",
            },
            Object {
              "id": 4,
              "team1": "C Palace",
              "team2": "Liverpool",
            },
            Object {
              "id": 5,
              "team1": "Brighton",
              "team2": "Leicester",
            },
            Object {
              "id": 6,
              "team1": "C Palace",
              "team2": "Liverpool",
            },
            Object {
              "id": 7,
              "team1": "Brighton",
              "team2": "Leicester",
            },
          ]
        }
        groupNumber={8}
        introHeadline="EFL CHAMPIONSHIP PREDICTOR"
        introTxt="Predict Home Win, Draw or Away Win to see who will finish in those crucial Promotion and Relegation places."
        onConfirm={[Function]}
        onLoaded={[Function]}
        onPredict={[Function]}
        teams={
          Array [
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
              "draw": 9,
              "lose": 4,
              "played": 31,
              "points": 63,
              "team": "Liverpool",
              "win": 18,
            },
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
              "draw": 9,
              "lose": 15,
              "played": 31,
              "points": 30,
              "team": "C Palace",
              "win": 7,
            },
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
              "draw": 10,
              "lose": 12,
              "played": 30,
              "points": 34,
              "team": "Brighton",
              "win": 8,
            },
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
              "draw": 10,
              "lose": 10,
              "played": 30,
              "points": 40,
              "team": "Leicester",
              "win": 10,
            },
          ]
        }
      />,
      "loaded": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "fixtures": Array [
          Object {
            "id": 0,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 1,
            "team1": "Brighton",
            "team2": "Leicester",
          },
          Object {
            "id": 2,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 3,
            "team1": "Brighton",
            "team2": "Leicester",
          },
          Object {
            "id": 4,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 5,
            "team1": "Brighton",
            "team2": "Leicester",
          },
          Object {
            "id": 6,
            "team1": "C Palace",
            "team2": "Liverpool",
          },
          Object {
            "id": 7,
            "team1": "Brighton",
            "team2": "Leicester",
          },
        ],
        "groupNumber": 8,
        "introHeadline": "EFL CHAMPIONSHIP PREDICTOR",
        "introTxt": "Predict Home Win, Draw or Away Win to see who will finish in those crucial Promotion and Relegation places.",
        "onConfirm": [Function],
        "onLoaded": [Function],
        "onPredict": [Function],
        "teams": Array [
          Object {
            "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
            "draw": 9,
            "lose": 4,
            "played": 31,
            "points": 63,
            "team": "Liverpool",
            "win": 18,
          },
          Object {
            "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
            "draw": 9,
            "lose": 15,
            "played": 31,
            "points": 30,
            "team": "C Palace",
            "win": 7,
          },
          Object {
            "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
            "draw": 10,
            "lose": 12,
            "played": 30,
            "points": 34,
            "team": "Brighton",
            "win": 8,
          },
          Object {
            "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
            "draw": 10,
            "lose": 10,
            "played": 30,
            "points": 40,
            "team": "Leicester",
            "win": 10,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Predictor
          fixtures={
            Array [
              Object {
                "id": 0,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 1,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 2,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 3,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 4,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 5,
                "team1": "Brighton",
                "team2": "Leicester",
              },
              Object {
                "id": 6,
                "team1": "C Palace",
                "team2": "Liverpool",
              },
              Object {
                "id": 7,
                "team1": "Brighton",
                "team2": "Leicester",
              },
            ]
          }
          groupNumber={8}
          introHeadline="EFL CHAMPIONSHIP PREDICTOR"
          introTxt="Predict Home Win, Draw or Away Win to see who will finish in those crucial Promotion and Relegation places."
          onConfirm={[Function]}
          onLoaded={[Function]}
          onPredict={[Function]}
          teams={
            Array [
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
                "draw": 9,
                "lose": 4,
                "played": 31,
                "points": 63,
                "team": "Liverpool",
                "win": 18,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
                "draw": 9,
                "lose": 15,
                "played": 31,
                "points": 30,
                "team": "C Palace",
                "win": 7,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
                "draw": 10,
                "lose": 12,
                "played": 30,
                "points": 34,
                "team": "Brighton",
                "win": 8,
              },
              Object {
                "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
                "draw": 10,
                "lose": 10,
                "played": 30,
                "points": 40,
                "team": "Leicester",
                "win": 10,
              },
            ]
          }
        />,
        "loaded": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fixtures": Array [
            Object {
              "id": 0,
              "team1": "C Palace",
              "team2": "Liverpool",
            },
            Object {
              "id": 1,
              "team1": "Brighton",
              "team2": "Leicester",
            },
            Object {
              "id": 2,
              "team1": "C Palace",
              "team2": "Liverpool",
            },
            Object {
              "id": 3,
              "team1": "Brighton",
              "team2": "Leicester",
            },
            Object {
              "id": 4,
              "team1": "C Palace",
              "team2": "Liverpool",
            },
            Object {
              "id": 5,
              "team1": "Brighton",
              "team2": "Leicester",
            },
            Object {
              "id": 6,
              "team1": "C Palace",
              "team2": "Liverpool",
            },
            Object {
              "id": 7,
              "team1": "Brighton",
              "team2": "Leicester",
            },
          ],
          "groupNumber": 8,
          "introHeadline": "EFL CHAMPIONSHIP PREDICTOR",
          "introTxt": "Predict Home Win, Draw or Away Win to see who will finish in those crucial Promotion and Relegation places.",
          "onConfirm": [Function],
          "onLoaded": [Function],
          "onPredict": [Function],
          "teams": Array [
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Liverpool.png?strip=all&w=67",
              "draw": 9,
              "lose": 4,
              "played": 31,
              "points": 63,
              "team": "Liverpool",
              "win": 18,
            },
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/CPFC-Primary-Badge.png?strip=all&w=72",
              "draw": 9,
              "lose": 15,
              "played": 31,
              "points": 30,
              "team": "C Palace",
              "win": 7,
            },
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2017/05/1024px-brighton__hove_albion_logo-svg_.png?strip=all&w=96",
              "draw": 10,
              "lose": 12,
              "played": 30,
              "points": 34,
              "team": "Brighton",
              "win": 8,
            },
            Object {
              "badge": "https://www.thesun.co.uk/wp-content/uploads/2016/04/Leicester-City_RGB.png?strip=all&w=90",
              "draw": 10,
              "lose": 10,
              "played": 30,
              "points": 40,
              "team": "Leicester",
              "win": 10,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
